exes/ethercat/inc/public/tfc/ec/devices/beckhoff/EQ2339.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <array>
+#include <bitset>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <span>
+#include <string_view>
+#include <system_error>
+#include <vector>
+
+#include <fmt/format.h>
+#include <boost/asio.hpp>
+#include <tfc/ec/devices/base.hpp>
+#include <tfc/ipc.hpp>
+
+namespace tfc::ec::devices::beckhoff {
+
+namespace asio = boost::asio;
+
+template <typename manager_client_type, template <typename, typename> typename signal_t = ipc::signal>
+class eq2339 final : public base {
+public:
+  static constexpr size_t size = 16;
+  static constexpr std::string_view name{ "EQ2339_0022" };
```

Comment:
```suggestion
  static constexpr std::string_view name{ "EQ2339" };
```
22 is irrelevant, https://www.beckhoff.com/en-en/products/i-o/ethercat-box/eqxxxx-stainless-steel-housing/eq23xx-digital-combi/eq2339-0022.html

Reply:
please make description why this is resolved and not implemented (in general).  History of conversation onsite does not reflect well online, so you can briefly tell why.

i did small research, https://www.beckhoff.com/en-en/download/440902134 beckhoff devices with subfix in product name share the same product code so when/if beckhoff creates 0021 it is most likely going to share the same product code, meaning this will be invalid.

Reply:
will do, the reason is the one above, having the 0022 suffix is more descriptive for the device and apt to include as to not confuse with the "EQ2339_0021" or other devices which might be implemented later.

Reply:
Like I said above, just simplified, you won´t be able to distinguish 0022 and 0021 from product code so if you would plug in a 0021 it would show up as 0022 which is incorrect.